### PR TITLE
Validate correctness of input arguments.

### PR DESCRIPTION
**Pull request by bangerth**
We don't check the input of the consensus algorithm functions: ranks for targets need to be unique.

/rebuild

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/base/mpi_consensus_algorithms.templates.h [hunk 1]
```diff
@@ -150,7 +150,21 @@ namespace Utilities
       {
 #ifdef DEAL_II_WITH_MPI
         // 1)
-        targets              = this->process.compute_targets();
+        targets = this->process.compute_targets();
+        Assert(
```
[the remark this answers is not in the document]
Comment (bangerth) [reply include/deal.II/base/mpi_consensus_algorithms.templates.h:154]: Where to? The base class only has an abstract virtual function but no other functionality. Or do you mean to create a tool function that encapsulates the lambda function?

[the remark this answers is not in the document]
Comment (peterrum) [reply include/deal.II/base/mpi_consensus_algorithms.templates.h:154]: OK, let's keep it out of the base class. Maybe move the content to free function (internal namespace) that returns a bool.

**Comment by bangerth** [comment]
OK, I put the two occurrences of the lambda function into a free function in the file.

**Review by peterrum — APPROVED** [review]

File: include/deal.II/base/mpi_consensus_algorithms.templates.h [hunk 2]
```diff
@@ -39,6 +39,25 @@ namespace Utilities
...
+        bool
+        has_unique_elements(const std::vector<unsigned int> &targets)
+        {
+          std::vector<unsigned int> my_destinations = targets;
+          const unsigned int        n_destinations  = my_destinations.size();
+          std::sort(my_destinations.begin(), my_destinations.end());
+          my_destinations.erase(std::unique(my_destinations.begin(),
+                                            my_destinations.end()),
+                                my_destinations.end());
+          return (my_destinations.size() == n_destinations);
```
Comment (kronbichler) [inline include/deal.II/base/mpi_consensus_algorithms.templates.h:57]: Can't you apply the trick from the other PR,
```suggestion
          return std::adjacent_find(my_destinations.begin(),
                                            my_destinations.end()) == 
                                my_destinations.end();
```

Comment (bangerth) [reply include/deal.II/base/mpi_consensus_algorithms.templates.h]: I can! So updated.

**Comment by bangerth** [comment]
The failing check is the clang-tidy one in `fe_point_evaluation.h` that is unrelated to this patch.

**Pull request merged**